Mathlib/Tactic/Mabel.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2024 Sam Ezeh. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sam Ezeh
+-/
+import Mathlib.Tactic.NormNum
+import Mathlib.Util.AtomM
+import Mathlib.Tactic.Abel
+import Qq
+
+/-!
+# The `mabel` tactic
+
+Evaluate expressions in the language of multiplicative, commutative monoids and groups.
+
+-/
+
+namespace Mathlib.Tactic.MAbel
+open Lean Elab Meta Tactic Qq Expr PrettyPrinter
+
+initialize registerTraceClass `mabel
+initialize registerTraceClass `mabel.detail
+
+variable {α : Type*}
+
+lemma Additive_identity : Additive α = α := rfl
+
+lemma ofMul_identity (x : α) : Additive.ofMul x = x := rfl
+
+/-- Tactic for converting equations in the language of
+multiplicative, commutative monoids and groups to those in the language
+of additive, commutative monoids and groups.
+-/
+syntax (name := to_additive) "to_additive" : tactic
+
+/-- The `to_additive` tactic, which solves converts equations in the language of
+multiplicative commutative additive groups (or monoids) to those in additive structures. -/
+elab_rules : tactic | `(tactic| to_additive) => withMainContext do
```

Comment:
This will need some improvement. `delab` seems the wrong approach...

Regarding `to_additive at h`, I think the skeleton would look like this:

```lean
syntax (name := to_additive) "to_additive" (location)? : tactic

elab_rules : tactic | `(tactic| to_additive $[$loc]?) => withMainContext do
  let loc := (loc.map expandLocation).getD (.targets #[] true)
  withLocation loc
    (atLocal := fun h ↦ do
      pure ()
    )
    (atTarget := do
      pure ()
    )
    (failed := fun _ ↦ throwError "to_additive made no progress")
```